components/brave_extension/extension/brave_extension/content_element_picker.ts

```diff
@@ -264,11 +275,17 @@ const quitElementPicker = () => {
   document.removeEventListener('scroll', elementPickerViewportChanged)
 }
 
-const launchElementPicker = () => {
-  pickerFrame = document.createElement('iframe')
+const attachElementPicker = async () => {
   // "src" is a web accessible resource since the URI is chrome-extension://.
   // This ensures a malicious page cannot modify the iframe contents.
-  pickerFrame.src = chrome.runtime.getURL('elementPicker.html')
+  pickerDiv = document.createElement('div')
+  const shadowRoot = pickerDiv.attachShadow({ mode: 'open' })
```

Comment:
We could still use an iframe and inject to the iframe's document via `iframe.contentWindow.document`, but I don't think there's really any disadvantage/advantage between shadow DOM and iframe - if the document really wants to observe this and interfere then they could either way. Actually if you make this a 'closed' shadow, then the main world wouldn't be able to access the element's contents, and they can't monkey patch `attachShadow` in this isolated world, right? So if you change this to 'closed', then it could be more protected.

Reply:
thanks for the suggestion, changed to "closed" shadow mode. 
BTW, `window` objects are unique in isolated world, so you can't just patch something (e.g. XHR request prototype) in the main world to affect isolated worlds.

Reply:
Yes but in a iframe or an open shadow, they could access the DOM elements and manipulate them. Less easy in a closed since they can't patch, so this is great.